### ISSUE TITLE
Not working anymore

### ISSUE THREAD

**Issue by jjnilton** [jomo/tld_checker#1]
Hi, thanks for sharing the script. Unfortunately, it's not working, tried on Ubuntu 17.10 and Debian 9. Seem's it's something in this part:

```
workers=32
for i in $(seq $((workers-1)) 0); do
  for tld in $(echo "$tlds" | awk "(NR+$i) % $workers == 0"); do
    check_tld "${1}.${tld}"
  done &
done
```

I replaced this function with the one in the previous commit and it worked without color in output and there's a strange suffix `^[(B^[[m` after every line. I'm trying to figure out why, but I don't know much of bash.

Changing `setf` to `setaf` fixed the color issue for me.

```
function check_tld {
  if [ -z "$(dig +short SOA "$1" $dns)" ]; then
    echo "$(tput setf 2)${1}$(tput sgr0)"
  else
    echo "$(tput setf 4)${1}$(tput sgr0)" >&2
  fi
}
```

**Comment by jomo** [comment]
Is color the only problem you're having or is something else not working? If so, can you please specify what "it's not working" means?

**Comment by jjnilton** [comment]
"Not working" means t does not output the checked domain names:

```
~/Downloads/scripts/tld_checker$ ./tld_checker.sh -v jose
# Using DNS server 'jose'
# Downloading TLD list from https://data.iana.org/TLD/tlds-alpha-by-domain.txt ...
######################################################################## 100,0%
# Getting available '-v' domains for 1541 TLDs ...
# done
~/Downloads/scripts/tld_checker$ 
```

The color is the issue when replacing one part of the current script with the previous version. 

I replaced this:

```
workers=32
for i in $(seq $((workers-1)) 0); do
  for tld in $(echo "$tlds" | awk "(NR+$i) % $workers == 0"); do
    check_tld "${1}.${tld}"
  done &
done
```
with this:

```
for tld in $tlds; do
 check_tld "${1}.${tld}" &
 sleep 0.005
done
```
Then I got the output I expected, but without colors.

**Comment by jomo** [comment]
You're using a `-v` flag, which is not interpreted by tld_checker.

> ```
> # Using DNS server 'jose'
> # Getting available '-v' domains for 1541 TLDs ...
> ```

It tries to use `-v` as the domain and `jose` as the DNS server. However, it should still output garbage in that case.

Can you try with proper arguments after inserting `set -x` in line 3 and paste the output here?

**Comment by jjnilton** [comment]
My bad, I forgot that it was `-x` to debug on bash.

Proper arguments:
```
j@jj-desktop:~/Downloads/scripts/tld_checker$ bash tld_checker.sh jose
# Downloading TLD list from https://data.iana.org/TLD/tlds-alpha-by-domain.txt ...
######################################################################## 100,0%
# Getting available 'jose' domains for 1541 TLDs ...
# done
~/Downloads/scripts/tld_checker$ 
```
After inserting `set -x` in the third line:
```
~/Downloads/scripts/tld_checker$ bash tld_checker.sh jose
+ '[' -z jose ']'
+ dns=
+ '[' -n '' ']'
+ tld_url=https://data.iana.org/TLD/tlds-alpha-by-domain.txt
+ echo '# Downloading TLD list from https://data.iana.org/TLD/tlds-alpha-by-domain.txt ...'
# Downloading TLD list from https://data.iana.org/TLD/tlds-alpha-by-domain.txt ...
++ curl --progress-bar https://data.iana.org/TLD/tlds-alpha-by-domain.txt
++ grep -v '^#'
######################################################################## 100,0%
+ tlds='AAA
AARP
ABARTH
<...>
ZONE
ZUERICH
ZW'
+ count=1541
+ echo '# Getting available '\''jose'\'' domains for 1541 TLDs ...'
# Getting available 'jose' domains for 1541 TLDs ...
+ workers=32
++ seq 31 0
+ wait
+ echo '# done'
# done
~/Downloads/scripts/tld_checker$ 
```

**Comment by jomo** [comment]
```bash
+ workers=32
++ seq 31 0
+ wait
+ echo '# done'
# done
```

Seems like it's not executing the for loop. I think you might not have `seq` installed?

I fixed the color issue in 5750b9aca78e869e4d1de7f45935e032254929f5 and removed the use of `seq` in 736c02685bc13b6ca8aef01a85ff773d1c35757b. Does that fix it for you?

**Comment by jjnilton** [comment]
seq is installed:
```
$ seq --help
Usage: seq [OPTION]... LAST
  or:  seq [OPTION]... FIRST LAST
  or:  seq [OPTION]... FIRST INCREMENT LAST
Print numbers from FIRST to LAST, in steps of INCREMENT.

Mandatory arguments to long options are mandatory for short options too.
  -f, --format=FORMAT      use printf style floating-point FORMAT
  -s, --separator=STRING   use STRING to separate numbers (default: \n)
  -w, --equal-width        equalize width by padding with leading zeroes
      --help     display this help and exit
      --version  output version information and exit

If FIRST or INCREMENT is omitted, it defaults to 1.  That is, an
omitted INCREMENT defaults to 1 even when LAST is smaller than FIRST.
The sequence of numbers ends when the sum of the current number and
INCREMENT would become greater than LAST.
FIRST, INCREMENT, and LAST are interpreted as floating point values.
INCREMENT is usually positive if FIRST is smaller than LAST, and
INCREMENT is usually negative if FIRST is greater than LAST.
INCREMENT must not be 0; none of FIRST, INCREMENT and LAST may be NaN.
FORMAT must be suitable for printing one argument of type 'double';
it defaults to %.PRECf if FIRST, INCREMENT, and LAST are all fixed point
decimal numbers with maximum precision PREC, and to %g otherwise.

GNU coreutils online help: <http://www.gnu.org/software/coreutils/>
Full documentation at: <http://www.gnu.org/software/coreutils/seq>
or available locally via: info '(coreutils) seq invocation'
```

Colors now are green and blue, instead of green and red, replacing `4` with `1` makes it red instead of blue.

But now I get this output: 

```
$ ./tld_checker.sh jose
# Downloading TLD list from https://data.iana.org/TLD/tlds-alpha-by-domain.txt ...
######################################################################## 100,0%
# Getting available 'jose' domains for 1541 TLDs ...
dig: '' is not a legal name (unexpected end of input)
jose.AC
dig: '' is not a legal name (unexpected end of input)
jose.ACADEMY
dig: '' is not a legal name (unexpected end of input)
jose.ACCOUNTANT
dig: '' is not a legal name (unexpected end of input)
```
I removed the double quotes `"` after `SOA`, around `$` and `$dns` in line 16 and seemed to fix it.

**Comment by jomo** [comment]
Whoops. I was wondering why I didn't quote `$dns` and immediately noticed, but then accidentally committed the quotes. Also didn't notice `setaf` uses different numbers for colors. Both issues should be fixed in 974808603a298afa24da1d42f941932e1c242744 and c98d46a3962df62ed660e39a824d902ece02fa8c, respectively. Thanks for reporting!
